deepdrr/vol.py

```diff
@@ -159,11 +160,132 @@ def from_nifti(
+        offset = ds.SharedFunctionalGroupsSequence[0].PixelValueTransformationSequence[0].RescaleIntercept
+        scale = ds.SharedFunctionalGroupsSequence[0].PixelValueTransformationSequence[0].RescaleSlope
+
+        # make user aware that this is only tested on windows
+        if ds.Manufacturer != "SIEMENS":
+            logger.warning("Multi-frame loading has only been tested on Siemens Enhanced CT DICOMs."
+                           "Please verify everything works as expected.")
+
+        # read the 'raw' data array
+        raw_data = ds.pixel_array.astype(np.float32)
+        hu_values = raw_data * scale + offset
+
+        # move slice index axis to back (k, j, i) -> (j, i, k)
+        hu_values = hu_values.transpose((1, 2, 0))
+
+        '''
+        EXPLANATION
+        
+        According to dicom (C.7.6.3.1.4 - Pixel Data) slices are of shape (Rows, Columns) 
+        => must be (j, i) indexed if we define i == horizontal and j == vertical.
+        This is taken care of in the definition of the affine transform 
+        Further reading: https://nipy.org/nibabel/dicom/dicom_orientation.html#i-j-columns-rows-in-dicom
+        '''
+
+        # transform the volume in HU to densities
+        data = load_dicom.conv_hu_to_density(hu_values)
+
+        # obtain materials analogous to nifti
+        if use_thresholding:
+            materials_path = cache_dir / f'{stem}_materials_thresholding.npz'
+            if use_cached and materials_path.exists():
+                logger.info(f'found materials segmentation at {materials_path}.')
+                materials = dict(np.load(materials_path))
+            else:
+                logger.info(f'segmenting materials in volume')
+                materials = load_dicom.conv_hu_to_materials_thresholding(hu_values)
+                np.savez(materials_path, **materials)
+        else:
+            materials_path = cache_dir / f'{stem}_materials.npz'
+            if use_cached and materials_path.exists():
+                logger.info(f'found materials segmentation at {materials_path}.')
+                materials = dict(np.load(materials_path))
+            else:
+                logger.info(f'segmenting materials in volume')
+                materials = load_dicom.conv_hu_to_materials(hu_values)
+                np.savez(materials_path, **materials)
+
+        # manually composing affine transform lps_from_ijk
+        # construct column for index k
+        k = np.array((last_slice_position - first_slice_position) / num_slices).reshape(3, 1)
+
+        # check if the calculated increment matches the SliceThickness (allow .1 millimeters deviations)
+        assert np.allclose(np.abs(k[2]), SliceThickness, atol=0.1, rtol=0)
+
+        # flip because dicom convention indexes a slice as [Columns, Rows]. see explanation above
+        CR = np.fliplr(RC)
+        CR_scaled = CR * PixelSpacing
+
```

Comment:
if the data was transposed to `(i, j, k)` internally, we don't need to flip here anymore.

Reply:
done.